modules/schema/src/main/java/org/apache/ignite/internal/schema/registry/UpgradingRowAdapter.java

```diff
@@ -160,7 +163,7 @@ public Byte byteValueBoxed(int colIdx) throws InvalidTypeException {
 
         Column column = mappedId < 0 ? mapper.mappedColumn(colIdx) : super.schema().column(mappedId);
 
-        if (NativeTypeSpec.INT8 != column.type().spec()) {
+        if (!isSupportedColumnTypeChange(column.type().spec().asColumnType(), ColumnType.INT8)) {
```

Comment:
```suggestion
       ensureColumnOfValidType(column.type().spec().asColumnType(), ColumnType.INT8);
```
Let's move this check to a separate method:
```
void ensureColumnOfValidType(ColumnType from, ColumnType to) {
    if (!isSupportedColumnTypeChange(from ,to)) {
        throw new SchemaException("Type conversion is not allowed: {} -> {}", from, to);
    }
}
```

Reply:
done